Spigot-Server-Patches/0419-Performance-improvement-for-Chunk.getEntities.patch

```diff
@@ -0,0 +1,37 @@
+From b9f79b55d41d863857ddec3b37bc148f2767b926 Mon Sep 17 00:00:00 2001
+From: wea_ondara <wea_ondara@alpenblock.net>
+Date: Thu, 10 Oct 2019 11:29:42 +0200
+Subject: [PATCH] Performance improvement for Chunk.getEntities
+
+This patch aims to reduce performance cost used by collecting the
+entities of a chunk. Previously the entitySlices were copied into an
+extra array with List.toArray() with is a costly and unneccessary
+operation. This patch will reduce the load of plugins which for example
+implement custom moblimits and depend on Chunk.getEntities().
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
+index b4551614..db758164 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
+@@ -110,14 +110,11 @@ public class CraftChunk implements Chunk {
+         Entity[] entities = new Entity[count];
+ 
+         for (int i = 0; i < 16; i++) {
+-
+-            for (Object obj : chunk.entitySlices[i].toArray()) {
+-                if (!(obj instanceof net.minecraft.server.Entity)) {
+-                    continue;
+-                }
+-
+-                entities[index++] = ((net.minecraft.server.Entity) obj).getBukkitEntity();
++            // Paper start - speed up (was with chunk.entitySlices[i].toArray() and cast checks which costs a lot of performance if called often)
++            for (net.minecraft.server.Entity entity : chunk.entitySlices[i]) {
```

Comment:
This potentially needs a null check, There have been issues in the past around null entries in random entity lists, I don't recall entirely if the entity slices where one of those common areas for this, but I sorta wanna say that it is

Reply:
added